### PR TITLE
Feat/datatransfer readme

### PR DESCRIPTION
**Pull request by shannonwells**
Adds a README for datatransfer, puts the original readme content into DESIGNDOC, adds a repo-level CONTRIBUTING guide, and adds some more content to the main README.

### CODE REVIEW

File: datatransfer/README.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
+# go-fil-components/datatransfer
+
+A go module to perform data transfers over [ipfs/go-graphsync](https://github.com/ipfs/go-graphsync)
+
+[![](https://img.shields.io/badge/made%20by-Protocol%20Labs-blue.svg?style=flat-square)](http://ipn.io)
+
+## Table of Contents
+* [Background](https://github.com/filecoin-project/go-fil-components#background)
+* [Usage](https://github.com/filecoin-project/go-fil-components#usage)\
```
Comment (hannahhoward) [inline datatransfer/README.md]: extra slash here (at EOL)

File: datatransfer/README.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    }
+```
+    
+For more detail, please see the [unit tests](https://github.com/filecoin-project/go-fil-components/blob/master/datatransfer/impl/graphsync/graphsync_impl_test.go).
+
+### Open a Push or Pull Request
+For a push or pull request you need a context, a `datatransfer.Voucher`, a host `peer.ID`, a base `cid.CID`
+and a selector.
+```go
+    func NewGraphsyncDatatransfer(ctx context.Context, h host.Host, gs graphsync.GraphExchange) {
```
Comment (hannahhoward) [inline datatransfer/README.md]: probably get rid the containing code (assume they read the instructions above), just list call to push/pull

File: datatransfer/README.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        dt := datatransfer.NewGraphSyncDataTransfer(h, gs)
+        channelID, err := dt.OpenPullDataChannel(ctx, host2.ID(), &voucher, baseCid, selector)
+    }
+```
+
+### Subscribe to Events
+
+
```
Comment (hannahhoward) [inline datatransfer/README.md]: for sure need to add this one... :)

File: datatransfer/README.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+## Contribute
+PRs are welcome!  Please first read the design docs and look over the current code.  PRs against 
+master require approval of at least two maintainers.  For the rest, please see our 
+[CONTRIBUTING](https://github.com/filecoin-project/go-fil-components/CONTRIBUTING.md) guide.
+
+## License
+This library is dual-licensed under Apache 2.0 and MIT terms.
```
Comment (hannahhoward) [inline datatransfer/README.md]: No need -- implied by license at root I think ?

File: README.md [hunk 5]
```diff
@@ -1 +1,9 @@
 # go-fil-components
+
+This repo houses modules for use by implementations of filecoin in go.
+
+## Components
+
+* [datatransfer](https://github.com/filecoin-project/go-fil-components/datatransfer), a module that enables sending data over 
+[libp2p](https://github.com/libp2p) using [ipfs/go-graphsync](https://github.com/ipfs/go-graphsync)
+* [filestore](https://github.com/filecoin-project/go-fil-components/filestore), ...
```
Comment (hannahhoward) [inline README.md]: add license section here

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# Contributing to this repo
+
+First, thank you for your interest in contributing to this project! Before you pick up your first issue and start
+changing code, please:
+ 
+1. review all documentation for the module you're interested in,
```
Comment (mishmosh) [inline CONTRIBUTING.md]: Capitalize R for consistency?

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
+# Contributing to this repo
+
+First, thank you for your interest in contributing to this project! Before you pick up your first issue and start
+changing code, please:
+ 
+1. review all documentation for the module you're interested in,  
+1. Look through the [issues for this repo,](https://github.com/filecoin-project/go-fil-components/issues)
```
Comment (mishmosh) [inline CONTRIBUTING.md]: "for any relevant discussions"?

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+1. Contact one of the maintainers for guidance.
```
Comment (mishmosh) [inline CONTRIBUTING.md]: A new reader probably won't know who maintainers are. How about "Write up a brief proposal in an issue."?

File: datatransfer/README.md (hunk 4 above)
Comment (mishmosh) [reply datatransfer/README.md]: Yup!

**Pull request merged**